src/main/java/core/basesyntax/dao/jdbc/ManufacturerDaoJdbcImpl.java

```diff
@@ -0,0 +1,118 @@
+package core.basesyntax.dao.jdbc;
+
+import core.basesyntax.dao.ManufacturerDao;
+import core.basesyntax.exceptions.DataProcessingException;
+import core.basesyntax.lib.Dao;
+import core.basesyntax.model.Manufacturer;
+import core.basesyntax.util.ConnectionUtil;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class ManufacturerDaoJdbcImpl implements ManufacturerDao {
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        String createString = "INSERT INTO manufacturers (manufacturer_name, "
+                + "manufacturer_country) VALUES (?,?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(createString,
+                        Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(1, manufacturer.getName());
+            statement.setString(2, manufacturer.getCountry());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                manufacturer.setId(resultSet.getObject(1, Long.class));
+            }
+            resultSet.close();
```

Comment:
it is optional to close ResultSet, because when a Statement object is closed, its current ResultSet object, if one exists, is also closed.
https://docs.oracle.com/en/java/javase/11/docs/api/java.sql/java/sql/Statement.html#close()

Reply:
done